smartsim/launchable/job.py

```diff
@@ -158,3 +159,35 @@ def __str__(self) -> str:  # pragma: no cover
         string = f"SmartSim Entity: {self.entity}\n"
         string += f"Launch Settings: {self.launch_settings}"
         return string
+
+
+@t.final
+class Record:
+    """A record a job that was launched and launch ID assigned to the
+    launching.
+    """
+
+    def __init__(self, launch_id: LaunchedJobID, job: Job) -> None:
+        """Initialize a new record of a launched job
+
+        :param launch_id: A unique identifier for the launch of the job.
+        :param job: The job that was launched.
+        """
+        self._id = launch_id
```

Comment:
Why are you deep copying the Job but not the launch_id here?

Reply:
In theory, the `launch_id` (at least for now), is just a string that we ask the type checker to treat like a different type, and strings are immutable in python, so a "deepcopy of" and a "reference to" a string are functionally equivalent.

There is actually some special logic in [`copy.deepcopy`](https://github.com/python/cpython/blob/3.13/Lib/copy.py#L119-L137) so that, by default, [it will just return the original reference to a string](https://github.com/python/cpython/blob/3.13/Lib/copy.py#L173-L183) if its encounter without change.

TL;DR I was lazy and didn't want to spell out "deepcopy". More than willing to add it though if we think that `LaunchedJobID` might become a more substantive type in future!